### ISSUE TITLE
Add authentification component

### ISSUE THREAD

**Issue by herzogrh** [citysciencelab/urban-model-platform#5]
There should be an authentification component that handles
1. **authentification of the model platform with different model servers.** Currently only BasicAuth is supported, but other model servers should be connectable via different authentification solutions (eg. Bearer Tokens, Digest Auth, ...)
2. authentification of users connecting to the model platform

There should be a separate identity and access management system (Keycloak) that is connected with the platform.

**Comment by hwbllmnn** [comment]
The second part is solved with #27

**Comment by hwbllmnn** [comment]
Regarding the first part, do we have any example servers with different authentication methods to test with?

**Comment by herzogrh** [comment]
Not yet unfortunately

**Comment by hwbllmnn** [comment]
@herzogrh Can this issue be closed? Maybe we can reopen this or open a new ticket once new model servers with different authentication methods pop up.

**Comment by hwbllmnn** [comment]
Note #47 regarding anonymous process access.

**Comment by hwbllmnn** [comment]
It should be better specified what exactly may happen for anonymous users, e.g. if they run processes, will job results be stored indefinitely? Will layers be created in Geoserver and be stored indefinitely? @herzogrh

**Comment by herzogrh** [comment]
Yes I think that's a good solution. I suppose only models which do not require a lot of computational power would be available to unauthorized users, so they would typically receive the results quite fast. Given that, it would be the easiest way if they'd have to wait for their current job, but not see any other jobs that were run by unauthorized users.

**Comment by hwbllmnn** [comment]
How do we decide which models do not require a lot of computational power, should this not be configurable?

**Comment by herzogrh** [comment]
Yes, this should be configured
